ossdbtoolsservice/query_execution/query_execution_service.py

```diff
@@ -357,9 +357,15 @@ def _execute_query_request_worker(self, worker_args: ExecuteRequestWorkerArgs):
 
         # Wrap execution in a try/except block so that we can send an error if it fails
         try:
-            query.execute(worker_args.connection)
+            query.execute(worker_args.connection, retry)
         except Exception as e:
-            self._resolve_query_exception(e, query, worker_args)
+            if not retry and worker_args.connection.connection.broken:
+                self._resolve_query_exception(e, query, worker_args, False, True)
+                conn = self._get_connection(worker_args.owner_uri, ConnectionType.QUERY)
```

Comment:
Curious if there's no need to set_connection in this case?

Reply:
Since this is being set for worker_args we can just set in line 365. The connection found in server from the session however took creating a setter to update the property.